interfaces/console/src/app/console/sites/page.tsx

```diff
@@ -152,36 +137,16 @@ export default function Page() {
         const siteId = data.getSiteStat.metrics[0].siteId;
         const metrics = data.getSiteStat.metrics;
 
-        // Process battery metrics
         metrics.forEach((metric) => {
           if (metric.type === 'battery_charge_percentage') {
-            const batteryLevel = metric.value;
-
-            setSitesStatus((prev) => {
-              const currentStatus = prev[siteId] || {
-                status: SITE_STATUS.ONLINE,
-                batteryStatus: 'Charged',
-                signalStrength: 'Strong',
-              };
-
-              let batteryStatus = 'Charged';
-              let newStatus = currentStatus.status;
-
-              if (batteryLevel < 20) {
-                batteryStatus = 'Low';
-                if (newStatus !== SITE_STATUS.OFFLINE) {
-                  newStatus = SITE_STATUS.WARNING;
-                }
-              } else if (batteryLevel < 50) {
-                batteryStatus = 'Medium';
-              }
+            setSiteMetrics((prev) => {
```

Comment:
why we need to store metrics in an other state? why not pass the data object directly?
Are you doing any computation on that?

Reply:
We store siteMetrics separately to merge data from multiple sources efficiently and keep the UI updates smooth. This avoids complex merging logic and ensures a responsive interface.

Reply:
I think by adding state in between you have to manage state on component change, which is more complex. Also its seems like unnecessary. 
We have Pubsub behaviour integrated in LineChart which get triggered when new data came and update the chart without page UI refresh. When we add the state a page level, any change in state will re-render whole page.